File: ruby/matrix_operations.rb
class MatrixOperations
    def add(a, b)
        a + b
    end
    
    def subtract(a, b)
        a - b
    end
    
    def scalar_multiply(matrix, scalar)
        matrix * scalar
    end

    def transpose(matrix)
        matrix.transpose
    end
    
    def sum(matrix)
        matrix.sum
    end
    
    def multiply(a, b)
        a * b
    end
    
    def vector_dot_product(a, b)
        a.dot(b)
    end
    
    def vector_cross_product(a, b)
        a.cross(b)
    end
    
    def inverse(matrix)
        matrix.inverse
    end
end

File: julia/src/MatrixOperations.jl
module MatrixOperations
    import LinearAlgebra

    function add(a::Matrix, b::Matrix)
        a + b
    end
    
    function subtract(a::Matrix, b::Matrix)
        a - b
    end

    function scalar_multiply(matrix::Matrix, scalar::Int)
        matrix * scalar
    end
    
    function transpose(matrix::Matrix)
        LinearAlgebra.transpose(matrix)
    end
    
    function sum(matrix::Matrix)
        Base.sum(matrix)
    end
    
    function multiply(a::Matrix, b::Matrix)
        a * b
    end

    function vector_dot_product(a::Vector, b::Vector)
        LinearAlgebra.dot(a, b)
    end
    
    function vector_cross_product(a::Vector, b::Vector)
        LinearAlgebra.cross(a, b)
    end
    
    function inverse(matrix::Matrix)
        LinearAlgebra.inv(matrix)
    end
end
